### PR TITLE
Update appdata files tags

### PR DESCRIPTION
**Pull request by aklsh**
- Done as part of flatpak packaging
- Updated existing fields, and added new ones according to [Freedesktop AppStream specification](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#sect-Metadata-GenericComponent)

### CODE REVIEW

**Comment by rfuest** [comment]
Concerning the application ID: This PR uses `net.sourceforge.GTKWave` as the application ID, but the GSettings schema and macOS plist use `com.geda.gtkwave`. `com.geda.gtkwave` is a bad ID according to the flatpak guidelines (https://docs.flatpak.org/en/latest/conventions.html#application-ids), because `geda.com` isn't related to GTKWave or the gEDA project. The GTK4 port will use the application ID in more places than the current version, so it would be good if we could agree on one ID.

These are the candidates IMO:
* `net.sourceforge.GTKWave`
* `io.github.GTKWave`
  If the project is supposed to move from SF to GitHub.
* `org.geda_project.GTKWave`
  I'm not sure of the status of the gEDA project and the relationship between GTKWave and gEDA.

**Comment by gtkwave** [comment]
I'd say use github.  Eventually, I'll update the sourceforge site such that it will state that it exists only for historical reasons (e.g., if it's referenced in a paper or technical literature, etc.).
-Tony


    On Thursday, August 4, 2022, 10:48:15 AM EDT, Ralf Fuest ***@***.***> wrote:  
 
 


Concerning the application ID: This PR uses net.sourceforge.GTKWave as the application ID, but the GSettings schema and macOS plist use com.geda.gtkwave. com.geda.gtkwave is a bad ID according to the flatpak guidelines (https://docs.flatpak.org/en/latest/conventions.html#application-ids), because geda.com isn't related to GTKWave or the gEDA project. The GTK4 port will use the application ID in more places than the current version, so it would be good if we could agree on one ID.

These are the candidates IMO:
   
   - net.sourceforge.GTKWave
   - io.github.GTKWave   
If the project is supposed to move from SF to GitHub.
   - org.geda_project.GTKWave   
I'm not sure of the status of the gEDA project and the relationship between GTKWave and gEDA.

**Comment by rfuest** [comment]
Sounds good to me. But we might want to use `io.github.gtkwave.GTKWave`, which is suggested in the link I mentioned earlier. The `io.github.gtkwave` namespace could then also be used for other apps, e.g. `io.github.gtkwave.rtlbrowse`.

**Comment by gtkwave** [comment]
That sounds good to me.  

-Tony


    On Thursday, August 4, 2022, 12:19:20 PM EDT, Ralf Fuest ***@***.***> wrote:  
 
 


Sounds good to me. But we might want to use io.github.gtkwave.GTKWave, which is suggested in the link I mentioned earlier. The io.github.gtkwave namespace could then also be used for other apps, e.g. io.github.gtkwave.rtlbrowse.

**Comment by aklsh** [comment]
Changed to `io.github.gtkwave.GTKWave`

**Comment by rfuest** [comment]
The appdata file looks good for version 3.3.111.

But why is GTKWave on Flathub built from a recent GIT version and not the 3.3.111 tarball? This causes a mismatch between the version shown in GTKWave and the appdata shown on Flathub (Version 3.4.0 vs. 3.3.111). And there is no description of the 3.4.0 changes in the appdata.

**Comment by aklsh** [comment]
Oops. I didn't know there was a 3.4.0 release. I thought 3.3.111 was the latest and used that. I've fixed it by using a 3.3.111 tarball instead. Should be able to see it in a few hours after flathub signs and uploads it.

**Review by rfuest — CHANGES_REQUESTED** [review]
I did try the Flatpak version of GTKWave and it is mostly working as expected.

Except from a few nitpicks about the appdata I noticed two issues with the Flatpak version:
* I couldn't run the transaction example because of mismatched GLIBC versions.
  My host GLIBC is newer than the version in the GNOME 42 runtime, which caused the `transaction` executable to fail. I'm not sure what the best solution for this problem is, because it also extends to any other libraries which are used by a transaction filter executable that aren't available inside the sandbox. There are methods to execute commands on the host by using `flatpak-spawn` or a D-Bus interface, but that would require code changes and adding additional permissions. Compiling the filter as a static executable did also fix the problem in my case. At the moment I would suggest to just except this issue as a known limitation of the Flatpak version.
* Printing doesn't work correctly.
  Only printing to a file is available in the Flatpak version. Adding `socket=cups` with Flatseal fixed this issue. So this should probably be added to the YAML file.

File: gtkwave3-gtk3/share/appdata/gtkwave.appdata.xml [hunk 1]
```diff
@@ -1,20 +1,47 @@
...
-<p>
+    <p>
 The viewer supports both post-mortem viewing of VCD files and interactive viewing of VCD data.  Tcl scripting and callback capability allow for remote control by other applications.
 </p>
-</description>
-<url type="homepage">http://gtkwave.sourceforge.net/</url>
-<screenshots>
-<screenshot type="default">http://gtkwave.sourceforge.net/gtkwave-appdata.png</screenshot>
-</screenshots>
-<updatecontact>bybell@rocketmail.com</updatecontact>
```
[the remark this answers is not in the document]
Comment (aklsh) [reply gtkwave3-gtk3/share/appdata/gtkwave.appdata.xml:19]: Oh my bad. I was using my email for the update contact for the flatpak and removed it while putting it here.

**Comment by aklsh** [comment]
> At the moment I would suggest to just except this issue as a known limitation of the Flatpak version.

Sure. I'll mention it in the flatpak description.

> Only printing to a file is available in the Flatpak version. Adding `socket=cups` with Flatseal fixed this issue. So this should probably be added to the YAML file.

Done.

**Comment by rfuest** [comment]
One final thing: Could you also update the other two appdata files to the latest version? I would fix this myself, but I don't have the required permissions to push to your branch.

I've tested the latest version on Flathub and printing works now. It's great to see that GTKWave is now also available as Flatpak.

**Review by rfuest — APPROVED** [review]

**Pull request merged**